### PR TITLE
Bug/e UI 3621 collection search filter aa

### PR DESCRIPTION
**Pull request by adnanakgun**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-3621

### Change description ###

Introduced extra param to control collections inside the search block and fixed a bug

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/shared/components/palette/collection/write-collection-field.component.ts [hunk 1]
```diff
@@ -183,6 +181,10 @@ export class WriteCollectionFieldComponent extends AbstractFieldWriteComponent i
     }
   }
 
+  isSearchFilter() {
```
Comment (phillip-whitaker) [inline src/shared/components/palette/collection/write-collection-field.component.ts:184]: Can we have a return type?

File: src/shared/components/palette/collection/write-collection-field.component.ts [hunk 2]
```diff
@@ -45,8 +44,7 @@ export class WriteCollectionFieldComponent extends AbstractFieldWriteComponent i
 
   constructor(private dialog: MatDialog,
               private scrollToService: ScrollToService,
-              private profileNotifier: ProfileNotifier,
-              private createChecker: CollectionCreateCheckerService
+              private profileNotifier: ProfileNotifier
```
Comment (phillip-whitaker) [inline src/shared/components/palette/collection/write-collection-field.component.ts:47]: can these be readonly?

File: src/shared/services/request/request.options.builder.ts [hunk 3]
```diff
@@ -46,7 +46,7 @@ export class RequestOptionsBuilder {
       if (caseCriteria) {
         for (let criterion of Object.keys(caseCriteria)) {
           if (RequestOptionsBuilder.includeParam(caseCriteria[criterion])) {
-            params = params.set(RequestOptionsBuilder.FIELD_PREFIX + criterion, caseCriteria[criterion].trim());
+            params = params.set(RequestOptionsBuilder.FIELD_PREFIX + criterion, caseCriteria[criterion].trim ? caseCriteria[criterion].trim() : caseCriteria[criterion]);
```
Comment (phillip-whitaker) [inline src/shared/services/request/request.options.builder.ts:49]: Can we take the building of variables out of the function calls?

Comment (udaydenduluri33) [reply src/shared/services/request/request.options.builder.ts:49]: Agree with Phil, If you are doing that you could write some tests for that function in isolation. Thanks

**Review by udaydenduluri33 — APPROVED** [review]
LGTMa

**Review by phillip-whitaker — DISMISSED** [review]

**Review by udaydenduluri33 — DISMISSED** [review]
LGTMa

**Pull request closed without merging**